mastermind.rb

```diff
@@ -52,38 +52,42 @@ def play
 		current_turn = 1
 		display_board
 		guess(current_turn)
-		feedback(current_turn)
-		while current_turn < 12 && @solution != @board[current_turn]["guess"] #!winner(current_turn)
+		solu = @solution
+		feedback(current_turn, solu)
+		while current_turn < 12 && solu != @board[current_turn]["guess"] #!winner(current_turn)
 			current_turn += 1
 			display_board
 			guess(current_turn)
-			feedback(current_turn)
+			feedback(current_turn, solu)
 		end
 
-		if @board[current_turn]["guess"] == @solution
+		if @board[current_turn]["guess"] == solu
 			puts "Congratulations!!! You stopped the nuclear detonation!!"
 		else
 			puts "The bomb exploded!!!!"
 			puts "You Lose!"
 		end
 	end
 
-	def feedback(current_turn)
+	def feedback(current_turn, solution_arr_in)
```

Comment:
something is deleting elements in @solution

Reply:
This is going to be hard for me to troubleshoot without seeing the code run. I am assuming it is just not deleting where you expect it to delete, because you have the code to delete stuff from the array. You could perhaps create a new array and push those deleted items into a new array to see that the items which are being deleted are the ones that you expect. I feel like I remember that delete can be finicky at times.